frontend_new/src/components/Cards.tsx

```diff
@@ -8,40 +8,73 @@ import Typography from "@material-ui/core/Typography";
 import Grid from "@material-ui/core/Grid";
 import { customTheme } from "../themes/customTheme";
 import ListItem from "@material-ui/core/ListItem";
+import BookmarkBorderOutlinedIcon from "@material-ui/icons/BookmarkBorderOutlined";
+import BookmarkOutlinedIcon from "@material-ui/icons/BookmarkOutlined";
+import IconButton from "@material-ui/core/IconButton";
+import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
+import Paper from "@material-ui/core/Paper";
 
 type props = {
   web_link: string;
   primary: string;
   secondary: string;
+  id: integer;
+  handlebookmark: () => void;
+  bookmarks: array;
+  isVisible: boolean;
 };
 function gotoLink(url: string) {
   window.location.href = url;
 }
 
 export default function SimpleCard(props: props) {
+  const [bookmarked, setBookmarked] = React.useState(true);
+  function handlebookmark() {
+    props.handlebookmark(
+      props.web_link,
+      props.primary,
+      props.secondary,
+      props.id
+    );
+    if (props.id in props.bookmarks) {
```

Comment:
**[Question]** the `handlebookmark `function would set the article in bookmarks, and since you already have the `useEffect`, do you still need this if-else statement?

Reply:
I think React wasnt detecting change in the bookmarks prop as it is an object, so this is why there is double logic

Reply:
thus I have the useeffect run when change to  boolean prop

Reply:
Probably best to define a function to do this and just call it both places. It makes it hard to maintain otherwise.

Reply:
good point - could write a seperate function that just takes an ID and a object and checks if id is in object. Ill fix this up

Reply:
As commented on slack, think unnecessary to have a shared function
```
setBookmarked(props.id in props.bookmarks);
```